src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToStringTransformer.php

```diff
@@ -132,6 +132,14 @@ public function reverseTransform($value)
             throw new TransformationFailedException('Expected a string.');
         }
 
+        // handle seconds ignored by user's browser when seconds as single_text is 0
+        if ($this->parseFormat === 'H:i:s|') {
+            if (!preg_match('((?:(?:[0-1][0-9])|(?:[2][0-3])|(?:[0-9])):(?:[0-5][0-9])(?::[0-5][0-9]))', $value) &&
+                preg_match('((?:(?:[0-1][0-9])|(?:[2][0-3])|(?:[0-9])):(?:[0-5][0-9]))', $value)) {
```

Comment:
Yes, that is why the condition $ this-> parseFormat === 'H: i: s |'.

And, ok, these default values would apply then sent the form (within the SUBMIT events) ?, if so, would work.

The ultimate goal of this is to correct the omission of the latter by the browser when the field requires seconds ('with_seconds' => true), and the user enters 0 seconds, in this case the browser sends '02: 05', in instead of '02: 05: 00 'causing a validation error.